src/com/sun/jna/Pointer.java

```diff
@@ -654,36 +656,110 @@ public ByteBuffer getByteBuffer(long offset, long length) {
+        int indexOfNull = ws.indexOf('\0');
+        return indexOfNull < 0 ? ws : ws.substring(0, indexOfNull);
+    }
+
+    /**
+     * Copy native memory to a Java String. The encoding used is obtained form
+     * {@link Native#getDefaultStringEncoding()}.
+     *
+     * @param offset
+     *            byte offset from pointer to start reading bytes
      * @return the <code>String</code> value being pointed to
      */
     public String getString(long offset) {
         return getString(offset, Native.getDefaultStringEncoding());
     }
 
+    /**
+     * Copy native memory to a Java String. The encoding used is obtained form
+     * {@link Native#getDefaultStringEncoding()}.
+     *
+     * @param offset
+     *            byte offset from pointer to start reading bytes
+     * @param maxBytes
+     *            the maximum number of bytes to read
+     * @return the <code>String</code> value being pointed to, up to either a null
+     *         terminator or <code>maxBytes</code>
+     */
+    public String getString(long offset, int maxBytes) {
+        return getString(offset, maxBytes, Native.getDefaultStringEncoding());
+    }
+
     /**
      * Copy native memory to a Java String using the requested encoding.
      *
-     * @param offset byte offset from pointer to obtain the native string
-     * @param encoding the desired encoding
+     * @param offset
+     *            byte offset from pointer to obtain the native string
+     * @param encoding
+     *            the desired encoding
      * @return the <code>String</code> value being pointed to
      */
     public String getString(long offset, String encoding) {
         return Native.getString(this, offset, encoding);
     }
 
-    /** Read a native array of bytes of size <code>arraySize</code> from the
-        given <code>offset</code> from this {@link Pointer}.
-    */
+    /**
+     * Copy native memory to a Java String using the requested encoding.
+     *
+     * @param offset
+     *            byte offset from pointer to obtain the native string
+     * @param maxBytes
+     *            the maximum number of bytes to read
+     * @param encoding
+     *            the desired encoding
+     * @return the <code>String</code> value being pointed to, up to either a null
+     *         terminator or <code>maxBytes</code>
+     */
+    public String getString(long offset, int maxBytes, String encoding) {
+        // Fetch the maxBytes
+        byte[] data = this.getByteArray(offset, maxBytes);
+        // Convert to String
+        String s = "";
+        if (encoding != null) {
+            try {
+                s = new String(data, encoding);
+            } catch (UnsupportedEncodingException e) {
+                s = new String(data, Charset.defaultCharset());
+            }
+        }
+        // Truncate at null if necessary
+        int indexOfNull = s.indexOf('\0');
+        return indexOfNull < 0 ? s : s.substring(0, indexOfNull);
```

Comment:
Lines 745-756 can be replaced with `return Native.toString(data, encoding)`.

Reply:
Having copied the code I used from there, I should have realized this... 😮 

Thanks for the review.  I understand if it doesn't get merged, but I figured it was easier to show what I intended.